### DISCUSSION TITLE
Antenna selection in RSPdx

### DISCUSSION THREAD

**Discussion by jacter74** [jvde-github/AIS-catcher#326] [General]
Hello everyone.

I've recently been trying to get my station up and running using ais-catcher (BTW: a great tool). I'm using RSPdx and I'm wondering if there's a way to control which antenna the radio uses? From what I've tried, I think the default is Antenna A, but Antenna B has a Bias-Tee output, which can be useful sometimes. Is it possible to switch to Antenna B or even C using AIS Catcher, e.g. with an unofficial parameter or somehow directly using SDRPlay_api?

**Comment by jvde-github** [comment]
Hi, it was not possible (it was by default using Antenna A). But I have added an option:
```
AIS-catcher -gs ANTENNA B
```
Can you test and see if it works? I have it running but it is not connected to an antenna so difficult to verify. I also saw some other DX specific parameters like notch, biastee, etc. I can add these as well if useful.

**Comment by jacter74** [comment]
Thank you for your quick response.

I wanted to do a quick test, without switching antennas (it's already dark outside), just observing the number of reported ships, but it didn't work. I started AIS-Catcher exactly as you said:

`AIS-catcher -gs ANTENNA B`

I got the message:

`[ERROR] Invalid Device setting: "ANTENNA"`

Before I started, I of course updated AIS-Catcher by running the script:

`sudo bash -c "$(wget -qO- https://raw.githubusercontent.com/jvde-github/AIS-catcher/main/scripts/aiscatcher-install)"`

Before and after the update, the program version did not change, I still have the version:

`AIS-catcher (build Nov 16 2024) v0.60-308-g977ab96f`

Am I doing something wrong or did something else fail?

**Reply by jvde-github** [reply]
Weird, just tested the script and installed the latest
```
AIS-catcher (build Nov 18 2024) v0.60-312-g4f7b402d
```

Was there any error message?

**Reply by jacter74** [reply]
Totally my fault. When I first time installed AIS-catcher I didn't use a script but did everything manually. I had quite a mess in my system, e.g. two copies of AIS-catcher. I cleaned it up and now everything works as it should. Sorry for the inconvenience.

So, after quick tests, the new option seems to work correctly. Switching antennas programmatically, after about 10 minutes I had the following results:

Ant.A - SkyScan - about 90 ships
Ant.B - no antenna - about 20 ships
Ant.C - YouLoop - about 60 ships

That makes sense. If I have time, I will try to check it more thoroughly in the next few days and if I find any irregularities, I will let you know.

As for the other options you mentioned. The ability to enable bias-tee (on antenna B) can be very useful if for some reason we do not want to use the built-in LNA. In the coming days I will be intensively testing various hardware and software configurations to find my optimum. If I come across anything interesting, I will of course share the information here. In the meantime, thank you for your great support.
